grapher/barCharts/DiscreteBarChart.tsx

```diff
@@ -456,12 +468,34 @@ export class DiscreteBarChart
         })
     }
 
-    @computed private get sortedRawSeries() {
+    @computed get sortConfig(): SortConfig {
+        if (this.manager.sortConfig) return this.manager.sortConfig
+
+        return {
+            sortBy: SortBy.total,
+            sortOrder: SortOrder.desc,
+        }
+    }
+
+    @computed private get sortedRawSeries(): DiscreteBarItem[] {
         const raw =
             this.seriesStrategy === SeriesStrategy.entity
                 ? this.entitiesAsSeries
                 : this.columnsAsSeries
-        return sortBy(raw, (series) => series.row.value)
+
+        let sortByFunc: (item: DiscreteBarItem) => number | string
+        switch (this.sortConfig.sortBy) {
+            case SortBy.entityName:
+                sortByFunc = (item: DiscreteBarItem) => item.seriesName
+                break
+            case SortBy.total:
+            case SortBy.column: // we only have one yColumn, so total and column are the same
+                sortByFunc = (item: DiscreteBarItem) => item.row.value
+                break
+        }
+        const sortedSeries = sortBy(raw, sortByFunc)
+        if (this.sortConfig.sortOrder === SortOrder.desc) sortedSeries.reverse()
```

Comment:
Wondering: How does this sort undefined values? E.g. a column is missing a value for an entity.

Reply:
Hm. For `DiscreteBarChart` I don't have explicit handling for undefined values, since I think we just don't show bars without values there?

In `StackedDiscreteBarChart`, undefineds are treated like 0. If there are multiple rows with a missing column value then the sort order between those is deterministic, but not really anything more than that.
We could potentially use the entity name to break ties, but not sure if that's worth it?

Reply:
Nope I don't think that's worth it. Thanks!